src/lib.rs

```diff
@@ -268,14 +269,14 @@ mod tests {
         assert!(!role_id.is_empty());
 
         // now get a secret id for this approle
-        let res: EndpointResponse<HashMap<String, String>> = c
+        let res: EndpointResponse<HashMap<String, Value>> = c
```

Comment:
Changing this type will require a major version update and I'd rather be able to get this out as a patch level change

Reply:
Unfortunately, response payload is 
`{
    "request_id": "1203e722-4631-8a61-57e2-a81dac7b7c18",
    "lease_id": "",
    "renewable": false,
    "lease_duration": 0,
    "data": {
        "secret_id": "c51c27ff-5d8f-a906-b5ca-da3236039419",
        "secret_id_accessor": "1eaeb71d-a34e-88ca-e624-549f7c9adfa8",
        "secret_id_ttl": 0   <------------------------------------------------------------------- not a string
    }
}`

and HashMap<String,String> conversion fails (for Vault v1.12.0)

https://developer.hashicorp.com/vault/docs/auth/approle#via-the-api-1